cpp/marc_tools/single_marc_convert_keywords.cc

```diff
@@ -59,69 +59,65 @@ void ExtractSubfieldsForTag(const MARC::Record &record, const std::string &subfi
+            keywords_to_gnd_numbers_map->emplace(std::make_pair(subfield_value_a, record.getControlNumber()));
+    }
+    ExtractSubfieldsForTag(record, field_tag, subfield_codes, subfields);
+    if (subfields->size() > 1) {
+        const std::string key(StringUtil::Join(*subfields, " "));
+        keywords_to_gnd_numbers_map->emplace(std::make_pair(key , record.getControlNumber()));
+    }
+}
 
 
-void ReadInGndKeywords(MARC::Reader * const marc_reader, std::unordered_map<std::string, std::string> * const gnd_keywords)
+void ReadInGndKeywords(MARC::Reader * const marc_reader, std::unordered_map<std::string, std::string> * const keywords_to_gnd_numbers_map)
 {
     unsigned record_count(0);
 
-    while (MARC::Record record = marc_reader->read()) {
+    while (const MARC::Record record = marc_reader->read()) {
         ++record_count;
 
         std::vector<std::string> subfields;
-        ExtractSubfieldsForTag(record, "150", "agx", &subfields);
-        // making 2 entries: one for every single 'a' subfield and one for all subfields
-        if (subfields.size() == 1)
-            gnd_keywords->emplace(std::make_pair(subfields[0], record.getControlNumber()));
-        ExtractSubfieldsForTag(record, "100", "atg", &subfields);
-        ExtractSubfieldsForTag(record, "110", "agx", &subfields);
-        ExtractSubfieldsForTag(record, "111", "agx", &subfields);
-        ExtractSubfieldsForTag(record, "130", "agx", &subfields);
-        ExtractSubfieldsForTag(record, "151", "agx", &subfields);
-        // second entry for all subfields combined
-        if (subfields.size() > 1) {
-            const std::string key(StringUtil::Join(subfields, " "));
-            gnd_keywords->emplace(std::make_pair(key , record.getControlNumber()));
-        }
+        AddMainSubfieldAndCombinationsToGndKeywords(record, keywords_to_gnd_numbers_map, "150", "agx", &subfields);
+        AddMainSubfieldAndCombinationsToGndKeywords(record, keywords_to_gnd_numbers_map, "100", "atg", &subfields);
+        AddMainSubfieldAndCombinationsToGndKeywords(record, keywords_to_gnd_numbers_map, "110", "agx", &subfields);
+        AddMainSubfieldAndCombinationsToGndKeywords(record, keywords_to_gnd_numbers_map, "111", "agx", &subfields);
+        AddMainSubfieldAndCombinationsToGndKeywords(record, keywords_to_gnd_numbers_map, "130", "agx", &subfields);
+        AddMainSubfieldAndCombinationsToGndKeywords(record, keywords_to_gnd_numbers_map, "151", "agx", &subfields);
     }
     LOG_INFO("Processed " + std::to_string(record_count) + " MARC record(s).");
 }
 
 
 void FindEquivalentKeywords(std::unordered_map<std::string, std::string> const &keywords_to_gnd_numbers_map, std::unordered_set<std::string> const &keywords_to_compare, const std::string matches_output_file, const std::string no_matches_output_file) {
-    std::unordered_map<std::string, std::string> keyword_matches;
+    std::unordered_map<std::string, std::string> keyword_to_ppn_map;
```

Comment:
The pattern we use for maps is keys_to_values_map, that is keys and values are plurals. So please rename this to keywords_to_ppns_map.